SoftLayer/managers/hardware.py

```diff
@@ -878,11 +889,26 @@ def upgrade(self, instance_id, memory=None,
 
         if prices:
             if test:
-                self.client['Product_Order'].verifyOrder(order)
+                result = self.client['Product_Order'].verifyOrder(order)
             else:
-                self.client['Product_Order'].placeOrder(order)
-            return True
-        return False
+                result = self.client['Product_Order'].placeOrder(order)
+        return result
+
+    def get_maintenance_windows_id(self, location_id):
+        """Get the disks prices to be added or upgraded.
+
+        :param int location_id: Hardware Server location id.
+        :return int.
+        """
+        begin_date_object = datetime.datetime.now()
+        begin_date = begin_date_object.strftime("%Y-%m-%dT00:00:00.0000-06:00")
+        end_date_object = datetime.date.today() + datetime.timedelta(days=30)
+        end_date = end_date_object.strftime("%Y-%m-%dT00:00:00.0000-06:00")
+
+        result_windows = self.client['SoftLayer_Provisioning_Maintenance_Window'].getMaintenanceWindows(begin_date,
+                                                                                                        end_date,
+                                                                                                        location_id)
+        return result_windows[0]['id']
```

Comment:
Make sure you get a result back before doing this, otherwise that will generate an ugly python error if there is no `result_windows[0]['id']` for whatever reason. I'm assuming the API might throw an error if that is the case, but just to make sure lets add some checks here to ensure we don't generate an exception here.

```suggestion
        if  len(result_windows) > 0:
            return result_windows[0].get('id')
```

Reply:
Done.